egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
   this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
 
+  if [ "$adaptive_shrink" == "true" ]; then
+    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
```

Comment:
I think both my formula
 this_shrink=$(perl -e "print 1.0 - (($this_learning_rate / $ilr) * (1 - $shrink));")
which has an l2 regularization interpretation, and your formula
 this_shrink=$(perl -e "print $shrink + ($flr/($this_learning_rate/$this_num_jobs)) * (1 - $shrink)") 
which allows the shrinkage value to approach 1 (and might be suitable for larger data sets)
should be experimented with.  But I think your formula could be clarified by modifying the existing script to set a variable this_effective_lrate first, and then set a variable this_learning_rate afterwards; you can then use $this_effective_lrate in your formula instead of that ratio.